### PR TITLE
feat: add zoom addon

### PR DESCRIPTION
**Pull request by Mastersam07**
Zoom addon

### List of issues which are fixed by the PR
#512 

### Screenshots

https://github.com/widgetbook/widgetbook/assets/31275429/38d142ae-4e57-4842-998d-786d22338064

### Checklist

- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on [Discord].

[CLA]: https://docs.google.com/forms/d/e/1FAIpQLScuRfjUzENsLsmQgqZlGLxMKbFi7zuXoPARyXytoyQrq7ntUw/viewform
[Discord]: https://discord.com/invite/zT4AMStAJA

### CODE REVIEW

File: packages/widgetbook/lib/src/addons/zoom_addon/zoom_addon.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          name: 'Zoom',
+          initialSetting: initialZoom,
+        );
+
+  @override
+  List<Field> get fields => [
+        ZoomControlField(
+          name: 'zoom',
+          initialValue: initialSetting,
+        ),
```
Comment (YoussefRaafatNasry) [inline packages/widgetbook/lib/src/addons/zoom_addon/zoom_addon.dart:19]: The fields should reflect primitive types, or simple ones. So I am not a fan of creating `ZoomControlField`, I would suggest we use the existing `DoubleSliderField` instead.

**Review by YoussefRaafatNasry — CHANGES_REQUESTED** [review]

File: packages/widgetbook/lib/src/addons/zoom_addon/zoom_addon.dart (hunk 1 above)
Comment (Mastersam07) [reply packages/widgetbook/lib/src/addons/zoom_addon/zoom_addon.dart]: This PR has been updated accordingly.

**Comment by Mastersam07** [comment]
Changes made @YoussefRaafatNasry

**Review by YoussefRaafatNasry — APPROVED** [review]
Here we go! Another good PR by @Mastersam07 🎉 
Great job with this one!

As you know, I've added minor improvements to speed up the review process and match the code-style, let me know if you have any questions about why a specific change was done.

**Pull request merged**